toolkit/block-producer-metadata/pallet/src/lib.rs

```diff
@@ -221,21 +235,73 @@ pub mod pallet {
 
 			ensure!(is_valid_signature, Error::<T>::InvalidMainchainSignature);
 
-			if BlockProducerMetadataStorage::<T>::get(cross_chain_key_hash).is_none() {
-				T::Currency::hold(
-					&HoldReason::MetadataDeposit.into(),
-					&origin_account,
-					T::HoldAmount::get(),
-				)
-				.map_err(|_| Error::<T>::InsufficientBalance)?;
+			if BlockProducerMetadataOwners::<T>::get(cross_chain_key_hash).is_none() {
```

Comment:
there is [contains_key](https://paritytech.github.io/polkadot-sdk/master/frame_support/storage/trait.StorageMap.html#tymethod.contains_key) just in case in case we do not use the value, but I will recommend to aggregate the storages into one entity.